packages/ag-charts-enterprise/src/features/error-bar/errorBar.ts

```diff
@@ -100,36 +108,63 @@ export class ErrorBars
         dataModel: _ModuleSupport.DataModel<any, any, any>,
         processedData: _ModuleSupport.ProcessedData<any>
     ) {
-        const { yLowerKey, yUpperKey, nodeData } = this;
+        const { xLowerKey, xUpperKey, yLowerKey, yUpperKey, nodeData } = this;
         const xIndex = dataModel?.resolveProcessedDataIndexById(this.cartesianSeries, `xValue`).index;
+        const yIndex = dataModel?.resolveProcessedDataIndexById(this.cartesianSeries, `yValue`).index;
         const xScale = this.cartesianSeries.axes[ChartAxisDirection.X]?.scale;
         const yScale = this.cartesianSeries.axes[ChartAxisDirection.Y]?.scale;
         if (!xScale || !yScale || !yLowerKey || !yUpperKey || xIndex === undefined) {
             return;
         }
 
-        const convert = (scale: _Scale.Scale<any, any, any>, value: any) => {
-            const offset = (scale.bandwidth ?? 0) / 2;
-            return scale.convert(value) + offset;
-        };
-
         nodeData.length = processedData.data.length;
 
         for (let i = 0; i < processedData.data.length; i++) {
             const { datum, values } = processedData.data[i];
             const xDatum = values[xIndex];
+            const yDatum = values[yIndex];
 
             if (yLowerKey in datum && yUpperKey in datum) {
+                let xPoints = undefined;
+                if (this.cartesianSeries.type == 'scatter') {
+                    if (xLowerKey !== undefined && xLowerKey in datum && xUpperKey != undefined && xUpperKey in datum) {
+                        xPoints = this.calculatePoints(
+                            xScale,
+                            datum[xLowerKey],
+                            datum[xUpperKey],
+                            yScale,
+                            yDatum,
+                            yDatum
+                        );
+                    } else {
+                        throw new Error(
+                            `AG Charts - series type 'scatter' requires xLowerKey (= ${xLowerKey}) and xUpperKey (= ${xUpperKey}) to exist in all data points`
+                        );
```

Comment:
Nit: Since this is a data-validation style error, we should console.warn() about this and carry on if possible - `Logger.warnOnce()` should be the right choice here so we don't spam the user for every missing data-poiont, on every reprocessing of the data:
https://github.com/ag-grid/ag-charts/blob/c1b1c01dd577a516ee1eaeb816c2a5bc16b7747c/packages/ag-charts-community/src/util/logger.ts#L25

Reply:
Good idea. Thanks for the feedback. I've changed this ([25c55792e](https://github.com/ag-grid/ag-charts/pull/258/commits/25c55792e1bf51c29bbc38fd15d637095ab2087e))